core/trino-spi/src/main/java/io/trino/spi/metrics/CustomMetrics.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.metrics;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomMetrics
+{
+    public static final CustomMetrics EMPTY = new CustomMetrics();
+
+    private final Map<String, Metric> map;
+
+    public CustomMetrics(Map<String, Metric>... maps)
+    {
+        this(Arrays.asList(maps));
+    }
+
+    public CustomMetrics(Iterable<Map<String, Metric>> maps)
```

Comment:
What's the use case for taking multiple maps?

Reply:
I am using it to merge multiple custom metrics at `OperatorStats#add`, but it's probably not a good choice for a c-tor - so I have renamed it into a static method `Metrics#merge`: https://github.com/trinodb/trino/pull/6232/commits/b4f3406d2b18f7fe821baeae0d1cec3156b89700#diff-de6c1298bb6eabc28a83bbb6305b6de4f2900681b3dfab627a0d541e2e50230aR34